src/TendoPay/Gateway.php

```diff
@@ -67,24 +67,31 @@ function __construct() {
 	}
 
 	public function maybe_add_outstanding_balance_notice() {
-		$error = isset( $_GET['witherror'] ) ? htmlspecialchars( $_GET['witherror'] ) : '';
+		$witherror = isset( $_GET['witherror'] ) ? $_GET['witherror'] : '';
+		$errors = explode(':', $witherror);
+		$errors = is_array($errors) ? array_map('htmlspecialchars', $errors) : [];
+		$error = isset($errors[0]) ? $errors[0] : '';
+		$extra = isset($errors[1]) ? $errors[1] : '';
+
+//		\error_log(__FILE__ . ' with error: ' . $witherror);
 		switch ( $error ) {
 			case 'outstanding_balance':
 				$notice =
 					__(
 						"Your account has an outstanding balance, please repay your payment so you make an additional purchase.",
 						'tendopay'
 					);
+				wc_print_notice( $notice, 'error' );
 				break;
 			case 'minimum_purchase':
-				// TODO replace 1000 to contant
-				$notice = __( "There is a minimum purchase amount of 1000 PHP.", 'tendopay' );
+			case 'maximum_purchase':
+				// TODO: check limit through api without return message.
+				$notice = __( $extra, 'tendopay' );
+				wc_add_notice( $notice, 'error' );
+				wp_redirect( wc_get_cart_url() );
```

Comment:
are you sure about this?

I think you will loose the notice when redirecting. 

even if not, then it this case redirect might not work (or might not work properly) if you won't "die" the web application right after it.

Reply:
it is tested. Added die after wp_redirect. Thank you for your review in detail. I'll read the reference manual more carefully next time. If you have an opinion or better suggestion as a professional Wordpress developer, please let me know. I appreciate your any word.